docs/infrastructure/microk8s/install.md

```diff
@@ -0,0 +1,274 @@
+
+## Pre-requisites
+
+* a machine running Ubuntu 18.04 LTS version.
+  * Further versions might work but we recommend using an LTS release.
+* a remote session (ssh) onto your machine of choice.
+
+## Cookbook
+
+Assuming that you are starting from a blank slate, the first step is to install microk8s. Otherwise, please jump on to the [Configuring microk8s](###Configuring-microk8s) section.
+
+### Installing microk8s
+
+1. Ensure that you are working off the root folder: `cd /root`.
+2. Make sure the machine is up to date and also has the `nfs-common` package installed: `apt update && apt install nfs-common -y`.
+3. We must run configure the instance's list of hosts with localhost. This in turn will allow the correct resolution of the local microk8s cluster on the kubeconfig. We can do so by running `sed -i -e "1s/127.0.0.1 localhost.*/127.0.0.1 localhost $(hostname)/" /etc/hosts`.
+4. Traffic forwarding must be configured on the machine so that the local microk8s cluster pods can reach the internet. This can be done by running `iptables -P FORWARD ACCEPT` (more info on this subject can be [found here](https://github.com/ubuntu/microk8s/tree/1.12#my-pods-cant-reach-the-internet-but-my-microk8s-host-machine-can)).
+5. Choose the microk8s release channel you want to install from.
+   1. The complete list of channels can be [found here](https://microk8s.io/docs/setting-snap-channel).
+   2. Assuming we want to install from the `stable` release channel, we export the following environment variable: `export MICROK8S_CHANNEL=stable`.
+6. We are finally in a state where we can install microk8s. That can be done via the `snap` package manager with: `snap install microk8s --classic --channel ${MICROK8S_CHANNEL}
+export PATH=$PATH:/snap/bin
+echo "--allow-privileged=true" >> /var/snap/microk8s/current/args/kube-apiserver
+microk8s.start && microk8s.status --wait-ready`
+
+### Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+   1. `sudo microk8s.enable dns && sudo microk8s.status --wait-ready --timeout 180`
+   2. `sudo microk8s.enable storage && sudo microk8s.status --wait-ready --timeout 180`
+   3. `sudo microk8s.enable ingress &&  sudo microk8s.status --wait-ready --timeout 180`
+   4. This will enable dns, storage and ingress on your microk8s cluster.
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster: `microk8s.stop && microk8s.start && microk8s.status --wait-ready
+sleep 10`. Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+   1. `microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s`
```

Comment:
I think we can squeeze all these commands in one that wait for everything and return to the user.
In a manual installation process we do not need to have timeout ideally and just let the user wait.